Add first operational version of code to convert KS survey data into shipping format

AutumnVulpes opened the pull request:
Creation of first working version for code that takes in Kickstarter backer survey results and returns an output csv with the required format and information requested by shipping partner.

Code flow:
1) Load the KS backer survey csv, and create an intermediate dictionary with only the required information fields.
2) Using the dictionary, create a list of lists, where each list corresponds to an individual row in the final output csv, and each item is a cell with the required info (name, address, etc).
3) Creation of the output csv and its headers, and then writing in each row based on the above list.

To-Dos:

- [x] 1) Write in appropriate code comments
- [ ] 2) Code in proper unit testing
- [x] 3) Resolve issue with name data (Splitting of first and last names expect two variables)
- [x] 4) Creation of functions (One for dict creation, one for list creation, one for writing to output csv)
- [x] 5) Splitting of non responders into seperate csv

AutumnVulpes committed:
Change function names, correcting of functions

Change function names to be more apparent
Fix row_to_backer_dict

AutumnVulpes commented:
Update from the original PR:

1. Code has been functionalized with appropriate function names
2. Have set up separate outputs for survey responders and non-responders
3. Coding comments added.
4. Code issues resolved (Unnecessary variables, fixing of name split)

methylDragon commented on survey_csv_formatter.py:
```diff
@@ -0,0 +1,114 @@
+    return bool(row["Shipping Address 1"])
+
+# Function to create a dictionary entry of each backer's required info
+def create_backer_entry(ks_data_row):
```
Please fix name as discussed

AutumnVulpes replied:
https://github.com/AutumnVulpes/survey-csv-formatter/compare/e955525bde90dc05f99bc403de48217d719a13d7..de50cfc25f91a3f2070dcce442c25e01dfd9d326

methylDragon commented on survey_csv_formatter.py:
```diff
@@ -0,0 +1,114 @@
+
+# Function to convert above dictionaries into lists for csv writing
+# Each SKU ordered by each is a new item in list (so backer with SKU A and B is 2 items)    
+def create_csv_list(target_dict):
```
Same here, please fix name

AutumnVulpes replied:
https://github.com/AutumnVulpes/survey-csv-formatter/compare/e955525bde90dc05f99bc403de48217d719a13d7..de50cfc25f91a3f2070dcce442c25e01dfd9d326

methylDragon commented on survey_csv_formatter.py:
```diff
@@ -0,0 +1,114 @@
+shipping_list = create_csv_list(responder_dict)
+non_response_list = create_csv_list(non_response_dict)
+
+# Function to write responder and non-responder lists into csvs
```
Nit: Reading the code it's clear it's a function, no need to write "Function to".

See example docstring
https://www.programiz.com/python-programming/docstrings

methylDragon replied:
Same as above

AutumnVulpes replied:
https://github.com/AutumnVulpes/survey-csv-formatter/compare/e955525bde90dc05f99bc403de48217d719a13d7..de50cfc25f91a3f2070dcce442c25e01dfd9d326

methylDragon commented on survey_csv_formatter.py:
```diff
@@ -0,0 +1,114 @@
+
+    csvfile.close()
+
+# Creation of two output csv; responders and non-responders
```
Nit: Imperative form

methylDragon replied:
Same as above

AutumnVulpes replied:
https://github.com/AutumnVulpes/survey-csv-formatter/compare/e955525bde90dc05f99bc403de48217d719a13d7..de50cfc25f91a3f2070dcce442c25e01dfd9d326

methylDragon commented on survey_csv_formatter.py:
```diff
@@ -0,0 +1,114 @@
+import csv
+ks_response = csv.DictReader(open("ks_data.csv"))
```
Bonus: Take in a command line argument instead of hard coding it

AutumnVulpes committed:
Change function names

Fix function names as stated in review

AutumnVulpes committed:
Add command line input

Add functionality to take in a csv file path from the command line when code is executed
